.github/workflows/dev_deploy.yml

```diff
@@ -51,10 +51,33 @@ jobs:
           AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
       - name: Publish Image to ECR
         run: docker push $AM_CLEANUP_IMAGE_TAG
+  build_record_thumbnail_lambda:
```

Comment:
I do think we want each lambda to have a separate build step; the O(n) pipeline steps don't worry me since the build steps can (and do) run in parallel. As long as each deployable object in this repo gets its own image, this seems like the way to do things.
In considering this, though, I have come to the realization the *reason* each deployable object in this repo gets its own image is a misunderstanding on my part. The first time I started working on a lambda in this project (about a year ago) I read that the code of a lambda had a size limit of 50 MB. I was concerned that in Typescript a monolithic project might exceed this, so I divided `stela` into multiple packages that could be deployed separately using npm workspaces. As it turns out though, that 50 MB limit only applies if you upload your code via zip file; if you use docker images they can be up to 10 GB. So while I originally though that a single application capable or running any of our lambdas or our api server or anything else in the project was off the table, I think I was mistaken. That said, the decoupling benefits of the current approach may be worthwhile on their own merits.

Reply:
Fully agree with all of that analysis!